components/ResponsiveHash/index.tsx

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useState } from 'react'
+import type { ReactNode } from 'react'
+import NextLink from 'next/link'
+import Tooltip from 'components/Tooltip'
+import CopyBtn from 'components/CopyBtn'
+import OpenInNewIcon from 'assets/icons/open-in-new.svg'
+import styles from './styles.module.scss'
+
+const ResponsiveHash = ({
+  label,
+  labelTooltip = null,
+  leading = 8,
+  href = null,
+  isExternalLink = false,
+  ellipsisPosition = 'middle',
+  monoFont = true,
+  btnRight = null,
+  copyAlertText = '',
+}: {
+  label: string
+  labelTooltip?: string
+  leading?: number
+  href?: string
+  isExternalLink?: boolean
+  ellipsisPosition?: 'middle' | 'end'
+  monoFont?: boolean
+  btnRight?: 'copy' | 'openInNew' | ReactNode
+  copyAlertText?: string
+}) => {
+  const [isMobile, setIsMobile] = useState(false)
+  const [modifiedLabel, setModifiedLabel] = useState(label)
+
+  useEffect(() => {
+    if (window && window.matchMedia('(max-width: 1024px)').matches) {
+      setIsMobile(true)
+    }
+  }, [])
```

Comment:
`useMemo` could be used if the value is computed on mount, but `isMobile` won't be updated on resize if it's only set on mount

Reply:
maybe make it update on resize is better, I can add a resize event listener